### DISCUSSION TITLE
Is there a .PDF version of this documentation available somewhere?

### DISCUSSION THREAD

**Discussion by GrantRobertson** [freeplane/docs#157] [Q&A]
I have tried clicking the little printer icon in the upper right corner of the docs home page, [here](https://docs.freeplane.org/home.html). Unfortunately, all the links to sections of the documentation, within the generated .PDF, all point to pages on the website, rather than to simply another page within the PDF. I completely understand that this is because clicking the little printer icon merely displays a special https://docs.freeplane.org/print.html page, and then calls the browser's printer dialog. And, while I do get a PDF that is perfectly useable for reading, it is not as useable as it could be, if the links jumped to the appropriate location within the actual PDF. (Yes, I know that many of the links within the documentation point to elsewhere on the FreePlane website. I am not talking about those links.)

I have tried all the different PDF printer drivers that I have, within both Chrome and Edge, with the same result. I suspect that this is because, when the mdBook utility (used to generate the FreePlane documentation website from markdown files) generates the print.html page, it is not designed to cause the links within that page to simply go to locations within that print.html page. Instead, it just leaves all the hyperlinks pointing to the same places as in the primary documentation pages. I know, this is an issue to take up with the mdBook developers. I am only mentioning it here to make it clear that I know where the hyperlink issue is coming from. (You can check this yourself by going to https://docs.freeplane.org/print.html, searching for "Command-line options," and checking the link there. Rather than pointing to the paragraph just a few lines away, it points off to the primary location for the page about Command-line options. Again, I know this is an issue with the mdBook html generator.)

I have looked into finding a way to convert markdown files to PDF. Unfortunately, the way the FreePlane documentation markdown files are organized on GitHub is laid out for mdBook to use. And, there is no way to easily convert mdBook source files to PDF. It is listed as an feature request on the mdBook GitHub repository. If the markdown files in this repository were organized such that they could be a GitBook, then there are converters available. What it comes down to is that I would have to clone this repository, reorganize the files and create whatever manifest files that GitBook requires, publish it to my own repository, and **then** be able to convert that to a PDF. And, I apologize, but that is just a bit more work than I am willing to go through. Especially since I have no clue how GitBook works, or even how to use Git. 

I know there will be some who say that I should just be content with the web-based documentation. And, it's fine. But, you cannot mark up web-based documentation with questions and one's own comments. I just learn about a software product so much better if I can add my own notes as I go through the documentation.

I am not saying that someone needs to generate properly linked PDF files for me. I am merely asking if that has already been done, and I just can't find it. Or if there is some other method for me to easily generate my own, that I don't know of.

Thank you.

**Comment by macmarrum** [comment]
I don't know if anyone has ever attempted to export the docs to PDF.

You might give [Obsidian](https://obsidian.md/) a try. 
Once you download [Freeplane/docs repo](https://github.com/freeplane/docs/archive/refs/heads/main.zip) and unzip it, you can open the directory in Obsidian. You'll be able to navigate through the Markdown files and edit their content.
There's `Canvas` functionality, that allows to put content from many files / sites in one page.

Obsidian has `Export to PDF` in its menu. I'd be interested to learn how well it works for you.

You can also have a look at https://medium.com/@clemensjarnach/combine-export-multiple-obsidian-markdown-files-as-pdf-f70b85e390ae

**Comment by dpolivaev** [comment]
Have you tried https://docs.freeplane.org/print.html ? 
Close the bookmarks on the page if needed
